internal/controllers/gateway/gateway_controller.go

```diff
@@ -583,6 +585,18 @@ func (r *GatewayReconciler) reconcileUnmanagedGateway(ctx context.Context, log l
 		debug(log, gateway, "Determining listener configurations from Kong data-plane")
 		kongListeners, err = r.determineListenersFromDataPlane(ctx, svc, kongListeners)
 		if err != nil {
+			errNoReadyClient := dataplane.NoReadyGatewayClientsError{}
+			if errors.As(err, &errNoReadyClient) {
```

Comment:
If we don't use the error, we can:

```suggestion
			if errNoG := (&dataplane.NoReadyGatewayClientsError{}); errors.As(err, &errNoG) {
```

Reply:
Done.